card/lib/card/set_pattern.rb

```diff
@@ -124,12 +125,26 @@ def format_module_list klass
         end
 
         def anchor_codenames
-          @anchor_name.parts.map do |part|
+          anchor_parts.map do |part|
             part_id = Card.fetch_id part
             part_id && Card::Codename[part_id.to_i] || (return nil)
           end
         end
 
+        def anchor_parts
+          return [@anchor_name] unless anchor_parts_count > 1
+          parts = @anchor_name.parts
+          if parts.size <= anchor_parts_count
+            parts
+          else
```

Comment:
Got it. Can we add this comment:
"Handles cases where anchor is a compound card, eg `A+B+*self`"